test/extended/two_node/tnf_kubelet_disruption.go

```diff
@@ -0,0 +1,299 @@
+			return
+		}
+
+		// Track if node count is unexpected, but attempt cleanup anyway
+		nodeCountUnexpected := len(nodeList.Items) != 2
+		if nodeCountUnexpected {
+			framework.Logf("Warning: Expected 2 nodes but found %d - attempting cleanup anyway", len(nodeList.Items))
+		}
+
+		// Attempt cleanup if we have at least 1 node
+		if len(nodeList.Items) >= 1 {
+			// Use the last available node for cleanup commands (prefer second node if available)
+			cleanupNode := nodeList.Items[0]
+			if len(nodeList.Items) >= 2 {
+				cleanupNode = nodeList.Items[1]
+			}
+
+			g.By("Cleanup: Clearing any kubelet resource bans that may exist")
+			framework.Logf("Cleanup: Clearing all bans and failures for kubelet-clone resource using node %s", cleanupNode.Name)
+			cleanupErr := utils.RemoveConstraint(oc, cleanupNode.Name, "kubelet-clone")
+			if cleanupErr != nil {
+				framework.Logf("Warning: Failed to clear kubelet-clone resource during cleanup: %v (this is expected if no bans were active)", cleanupErr)
+			} else {
+				framework.Logf("Successfully cleared all bans and failures for kubelet-clone resource during cleanup")
+			}
+
+			g.By("Cleanup: Waiting for all nodes to become Ready after resource ban cleanup")
+			for _, node := range nodeList.Items {
+				// Use a non-blocking check with logging instead of assertion
+				ready := false
+				for i := 0; i < int(kubeletRestoreTimeout/kubeletPollInterval); i++ {
+					nodeObj, err := oc.AdminKubeClient().CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+					if err == nil && nodeutil.IsNodeReady(nodeObj) {
+						ready = true
+						framework.Logf("Node %s is Ready after cleanup", node.Name)
+						break
+					}
+					time.Sleep(kubeletPollInterval)
+				}
+				if !ready {
+					framework.Logf("Warning: Node %s did not become Ready within timeout after cleanup", node.Name)
+				}
+			}
+
+			g.By("Cleanup: Validating etcd cluster status after test cleanup")
+			// Use a non-blocking check with logging instead of assertion
+			etcdHealthy := false
+			for i := 0; i < int(kubeletRestoreTimeout/kubeletPollInterval); i++ {
+				if err := utils.LogEtcdClusterStatus(oc, "AfterEach cleanup", etcdClientFactory); err == nil {
+					etcdHealthy = true
+					framework.Logf("Etcd cluster is healthy after cleanup")
+					break
+				}
+				time.Sleep(kubeletPollInterval)
+			}
+			if !etcdHealthy {
+				framework.Logf("Warning: Etcd cluster did not become healthy within timeout after cleanup")
+			}
+		} else {
+			framework.Logf("Critical: Cannot perform cleanup - no nodes available")
+		}
+
+		// Log node count issue but don't fail - cleanup should always complete
+		if nodeCountUnexpected {
+			framework.Logf("Warning: Expected exactly 2 nodes for two-node cluster but found %d - cluster topology may have changed during test", len(nodeList.Items))
+		}
+	})
+
+	g.It("Should recover from single node kubelet service disruption", func() {
```

Comment:
Lets' lowerase these `Should`s, dont forget that this word is in the middle of the test tile.

"Should" -> "should"